components/contentselector/model/contentmodel.cpp

```diff
@@ -408,6 +408,17 @@ bool ContentSelectorModel::ContentModel::dropMimeData(const QMimeData *data, Qt:
 
 void ContentSelectorModel::ContentModel::addFile(EsmFile *file)
 {
+    for (int row = 0; row < mFiles.size(); row++)
+    {
+        if (!mFiles.at(row)->fileName().compare(file->fileName(), Qt::CaseInsensitive))
+        {
+            beginRemoveRows(QModelIndex(), row, row);
+                mFiles.removeAt(row);
```

Comment:
1. It seems you just remove file from list and do not delete it. It can lead to memory leaks.
2. Do we really need a such bulk construction to just replace a value in QList?

Reply:
To 2: the file in question is **not** "replaced", and it shouldn't be. There is a difference between a) removing an existing basically random item from the container and appending a new different one and b) replacing the item. And QList has no means to do a.